java/sage/Carny.java

```diff
@@ -1159,61 +1617,87 @@ private void stdProcessing()
     newWPMap.keySet().removeAll(watchedPotsToClear);
 
     if (Sage.DBG) System.out.println("CARNY Negative Energy Size: " + (blackBalled.size() + stoners.length));
-    for (Airing badAir : blackBalled)
+    for (int i = 0, blackBalledSize = blackBalled.size(); i < blackBalledSize; i++)
```

Comment:
I was cutting down on object creation which was a much bigger performance killer due to all of the GC. There are cases where they are faster and cases where they are slower. The JVM isn't always great optimizing the iterators and I got a boost from just stopping the use of them.